### PR TITLE
Update RELEASING.md

### PR DESCRIPTION
**Pull request by vincentschen**
## Description of proposed changes
* Include recent changes to process

## Checklist

* [x] I have read the **CONTRIBUTING** document.
* [x] I have updated the documentation accordingly.

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -16,11 +15,6 @@ index-servers =
...
 username=YOUR_USERNAME
 password=YOUR_PASSWORD
-
-[pypitest]
-repository=https://test.pypi.org/legacy/
-username=YOUR_USERNAME
-password=YOUR_PASSWORD
 ```
 
 Then run `chmod 600 ./pypirc` so only you can read/write.
```
Comment (henryre) [inline RELEASING.md]: `./.pypirc`

Comment (henryre) [reply RELEASING.md]: and below, remove thing about permissions for master push

File: RELEASING.md [hunk 2]
```diff
@@ -57,8 +51,11 @@ You'll also need to have permissions to push directly to the `master` branch.
        python3 setup.py sdist  # create a source distribution
        python3 setup.py bdist_wheel  # create a universal wheel
 
+1. Attach the resulting binaries in (`dist/snorkel-x.x.x.*`) to the release.
+
 1. Check that everything looks correct by uploading the package to the PyPI test server:
 
+       pip install twine  # if not installed
```
Comment (henryre) [inline RELEASING.md]: update conda instructions below to say PR is auto-submitted and a maintainer needs to accept

File: RELEASING.md [hunk 3]
```diff
@@ -57,8 +51,11 @@ You'll also need to have permissions to push directly to the `master` branch.
...
        twine upload dist/* -r pypitest  # publish to test.pypi.org
        python3 -m venv test_snorkel  # create a virtualenv for testing
        source test_snorkel/bin/activate  # activate virtualenv
```
Comment (henryre) [inline RELEASING.md]: update RTD instructions: trigger a build for master to pull new tags, go to versions tab and add new tag, go to admin/advanced to set new default version

File: RELEASING.md [hunk 4]
```diff
@@ -38,9 +32,9 @@ You'll also need to have permissions to push directly to the `master` branch.
 1. Make sure `CHANGELOG.md` is up to date for the release: compare against PRs
    merged since the last release & update top heading with release date.
 
-1. Update version to, e.g. 0.9.0 (remove the `+dev` label) in `snorkel/VERSION.py`.
+1. Update version to, e.g. 0.9.0 (remove the `+dev` label) in `snorkel/version.py`.
 
-1. Commit these changes and push to master:
+1. Commit these changes and create a PR:
```
Comment (henryre) [inline RELEASING.md]: below this block: merge the PR, pull master, then proceed to "Tag the release"

File: RELEASING.md [hunk 5]
```diff
@@ -38,15 +30,15 @@ You'll also need to have permissions to push directly to the `master` branch.
...
 
        git add . -u
        git commit -m "[RELEASE]: v0.9.0"
        git push origin master
 
-1. Tag the release:
+1. On approval, merge the PR, pull master locally, and proceed to "tag the release":
```
Comment (henryre) [inline RELEASING.md]: lolol ok if you're going to take me literally

```
...
    git commit -m "[RELEASE]: v0.9.0"
    git push origin master

1. Once the PR is approved, merge it and pull master locally.
1. Tag the release:
...
```

File: RELEASING.md [hunk 6]
```diff
@@ -94,7 +86,10 @@ You'll also need to have permissions to push directly to the `master` branch.
        git push origin master
        
 1. Add the new tag to [the Snorkel project on ReadTheDocs](https://readthedocs.org/projects/snorkel),
-   set it as the default version, and make sure a build is triggered.
+    * Trigger a build for master to pull new tags.
+    * Then, go to the "Versions" tab and add a tag.
```
Comment (henryre) [inline RELEASING.md]: ...tab and set the new tag to Active.

File: RELEASING.md [hunk 7]
```diff
@@ -68,10 +63,7 @@ You'll also need to have permissions to push directly to the `master` branch.
 
        twine upload dist/* -r pypi
 
-1. Fork [`conda-forge/snorkel-feedstock`](https://github.com/conda-forge/snorkel-feedstock),
-   update the version in
-   [`meta.yml`](https://github.com/conda-forge/snorkel-feedstock/blob/master/recipe/meta.yaml),
-   submit a PR upstream and follow conda-forge's PR instructions.
+1. A PR is auto-submitted on [`conda-forge/snorkel-feedstock`](https://github.com/conda-forge/snorkel-feedstock) to update the version. A maintainer needs to accept and merge those changes.
```
Comment (henryre) [inline RELEASING.md]: Nit: newline sentences

Comment (henryre) [inline RELEASING.md]: Add: It will take up to a couple hours for the PR to be submitted.

Comment (vincentschen) [reply RELEASING.md]: new bullet!

**Pull request merged**